rsmtool/utils/commandline.py

```diff
@@ -230,117 +253,620 @@ def existing_configuration_file(string):
     return parser
 
 
-def generate_configuration(context,
-                           use_subgroups=False,
-                           as_string=False,
-                           suppress_warnings=False):
+class InteractiveField:
     """
-    Automatically generate an example configuration for a given
-    command-line tool.
+    Class that encapsulates a configuration field that is computed interactively.
 
-    Parameters
+    Attributes
+    ----------
+    choices : list
+        List of possible choices for the field value if ``data_type`` is "choice".
+        An empty list for all other types of fields.
+    complete_style : prompt_toolkit.shortcuts.prompt.CompleteStyle
+        A CompleteStyle that defines how to style the completions.
+        Set to ``CompleteStyle.MULTI_COLUMN`` for fields that have ``data_type``
+        of "choice". Set to ``None`` for all other field types.
+    completer : prompt_toolkit.completions.base.Completer
+        A ``Completer`` object used to provide auto-completion for the field.
+        The actual completer used depends on the ``data_type`` of the field.
+        For example, for fields of type "choice", we use a ``FuzzyWordCompleter``,
```

Comment:
```suggestion
        For example, for fields of type ``choice``, we use a ``FuzzyWordCompleter``,
```

Now that I'm thinking about it actually, shouldn't it be ``"choice"``? It's the raw value that is within \`\`. And I think variables can be simply represented with one \`. But, it's not that important, anyway. Just thought I'd point it out. There are other instances in this file where the style could be enforced, but I won't point them all out.

Reply:
Yes you are right, it should be `"choice"`. Good catch. I will try and find all other instances an fix them too.